src/encoding.rs

```diff
@@ -1366,6 +1375,7 @@ mod test {
             &mut buf,
             DecodeContext::default(),
         ) {
+            use crate::alloc::string::ToString;
```

Comment:
Interesting, does no-std mean there's no prelude?  Separately, why is this using a re-export from `crate`?

Reply:
when using no_std you get the core prelude, but not the std prelude

https://doc.rust-lang.org/1.8.0/book/using-rust-without-the-standard-library.html

there is no `alloc` prelude, anything like `ToString` from alloc must be imported explicitly

Reply:
i don't honestly remember why this was `crate::alloc`, I'm going to change it to just `alloc::string::ToString`

Reply:
seems to work without the crate part

Reply:
Yeah, no need for `crate::alloc::*`, or `::alloc::*`, just `alloc` works

Reply:
Also there was an `alloc::prelude::v1` for awhile but it was removed prior to stabilization 😢